### ISSUE TITLE
Negative exponential of non-commutative symbol commutes

### ISSUE THREAD

**Issue by Adrien-Kahn** [sympy/sympy#27209]
When `a` and `b` are non-commutative symbols, `simplify` makes `sp.exp(-a)` and `b` commute even though the former is correctly identified as non-commutative:

```
import sympy as sp

a = sp.Symbol('a', commutative=False)
b = sp.Symbol('b', commutative=False)

print(sp.exp(-a).is_commutative)
print(b.is_commutative)

expr = sp.exp(-a) * b - b * sp.exp(-a)
print(expr)
print(expr.simplify())
```
Output:
```
False
False
-b*exp(-a) + exp(-a)*b
0
```

This is especially strange since dropping the minus sign restores the expected behaviour:
```
import sympy as sp

a = sp.Symbol('a', commutative=False)
b = sp.Symbol('b', commutative=False)

print(sp.exp(a).is_commutative)
print(b.is_commutative)

expr = sp.exp(a) * b - b * sp.exp(a)
print(expr)
print(expr.simplify())
```
Output:
```
False
False
-b*exp(a) + exp(a)*b
-b*exp(a) + exp(a)*b
```

**Comment by Adrien-Kahn** [comment]
I'm sorry, I had an outdated version of SymPy, the bug is fixed in the latest version.

**Comment by oscarbenjamin** [comment]
I bisected the fix to 2f6a55c98331d34b6f5196609309e6ec6a7c9624 from gh-25110. I believe this was fixed in SymPy 1.12.
